Dart/src/com/jetbrains/lang/dart/ide/hierarchy/DartHierarchyUtil.java

```diff
@@ -55,4 +60,8 @@ public static DartClass findDartClass(@NotNull final Project project, @NotNull f
 
     return DartAnalysisServerService.getInstance().search_getTypeHierarchy(file, name.getTextRange().getStartOffset(), false);
   }
+
+  public static boolean isTypeExecutable(IElementType type) {
+    return FUNCTION_DEFINITION.contains(type) || type == FACTORY_CONSTRUCTOR_DECLARATION || type == NAMED_CONSTRUCTOR_DECLARATION;
```

Comment:
I guess checking DartComponentType would be better. When PSI contains all info we need and the task can be solved at PSI level we try not to go down to ASTNode, IElementType, etc. 
The same in `DartCallHierarchyProvider`, I'd prefer using `instanceof` instead of `getTokenType()`.
Not a major thing, just a common practice.